### DISCUSSION TITLE
Compiled EXE is including unused libraries

### DISCUSSION THREAD

**Discussion by non-npc** [brentvollebregt/auto-py-to-exe#497] [General]
I am compiling my exe to a folder and I noticed in the _internal folder, it is putting libraries that I am not using  in my project.
For example it is adding pygame, which I am not even using which is what brought up this question in the first place.
How do I ensure the compiled output only includes the libraries it needs to function ?

For example this is my imports:

- `import webview
- import traceback
- import logging
- import sys
- import json
- import os
- import subprocess
- from urllib.parse import unquote
- from datetime import datetime
- from functools import wraps
- import threading
- import time
- import psutil`

and after compiling with auto-py-to-exe, 
in the output "_interal" folder this is what is in there:

- `2024-09-07  12:33 PM    <DIR>          aiohttp
- 2024-09-07  12:33 PM            22,112 api-ms-win-core-console-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-datetime-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-debug-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-errorhandling-l1-1-0.dll
- 2024-09-07  12:33 PM            26,216 api-ms-win-core-file-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-file-l1-2-0.dll
- 2024-09-07  12:33 PM            18,696 api-ms-win-core-file-l2-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-handle-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-heap-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-interlocked-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-libraryloader-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-localization-l1-2-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-memory-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-namedpipe-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-core-processenvironment-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-processthreads-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-processthreads-l1-1-1.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-profile-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-rtlsupport-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-string-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-synch-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-synch-l1-2-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-sysinfo-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-timezone-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-core-util-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-crt-conio-l1-1-0.dll
- 2024-09-07  12:33 PM            26,216 api-ms-win-crt-convert-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-crt-environment-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-crt-filesystem-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-crt-heap-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-crt-locale-l1-1-0.dll
- 2024-09-07  12:33 PM            30,328 api-ms-win-crt-math-l1-1-0.dll
- 2024-09-07  12:33 PM            75,368 api-ms-win-crt-private-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-crt-process-l1-1-0.dll
- 2024-09-07  12:33 PM            26,208 api-ms-win-crt-runtime-l1-1-0.dll
- 2024-09-07  12:33 PM            26,232 api-ms-win-crt-stdio-l1-1-0.dll
- 2024-09-07  12:33 PM            26,232 api-ms-win-crt-string-l1-1-0.dll
- 2024-09-07  12:33 PM            22,120 api-ms-win-crt-time-l1-1-0.dll
- 2024-09-07  12:33 PM            22,136 api-ms-win-crt-utility-l1-1-0.dll
- 2024-09-07  12:33 PM    <DIR>          attrs-23.2.0.dist-info
- 2024-09-07  12:33 PM         1,334,069 base_library.zip
- 2024-09-07  12:33 PM    <DIR>          bcrypt
- 2024-09-07  12:33 PM    <DIR>          cefpython3
- 2024-09-07  12:33 PM    <DIR>          certifi
- 2024-09-07  12:33 PM    <DIR>          cffi-1.16.0.dist-info
- 2024-09-07  12:33 PM    <DIR>          charset_normalizer
- 2024-09-07  12:33 PM    <DIR>          clr_loader
- 2024-09-07  12:33 PM    <DIR>          cryptography
- 2024-09-07  12:33 PM    <DIR>          cryptography-42.0.8.dist-info
- 2024-09-07  12:33 PM    <DIR>          dnspython-2.6.1.dist-info
- 2024-09-07  12:33 PM           654,336 freetype.dll
- 2024-09-07  12:33 PM    <DIR>          frozenlist
- 2024-09-07  12:33 PM    <DIR>          gevent
- 2024-09-07  12:33 PM    <DIR>          gevent-24.2.1.dist-info
- 2024-09-07  12:33 PM    <DIR>          greenlet
- 2024-09-07  12:33 PM    <DIR>          greenlet-3.0.3.dist-info
- 2024-09-07  12:33 PM    <DIR>          importlib_metadata-7.2.1.dist-info
- 2024-09-07  12:33 PM    <DIR>          jaraco
- 2024-09-07  12:33 PM         5,191,960 libcrypto-3.dll
- 2024-09-07  12:33 PM            39,696 libffi-8.dll
- 2024-09-07  12:33 PM         1,272,353 libjpeg-62.dll
- 2024-09-07  12:33 PM           244,224 libjpeg-9.dll
- 2024-09-07  12:33 PM           265,216 libmodplug-1.dll
- 2024-09-07  12:33 PM            35,328 libogg-0.dll
- 2024-09-07  12:33 PM           370,176 libopus-0.dll
- 2024-09-07  12:33 PM            51,200 libopusfile-0.dll
- 2024-09-07  12:33 PM           279,040 libpng16-16.dll
- 2024-09-07  12:33 PM           787,224 libssl-3.dll
- 2024-09-07  12:33 PM           387,584 libtiff-5.dll
- 2024-09-07  12:33 PM           175,616 libwavpack-1.dll
- 2024-09-07  12:33 PM           444,416 libwebp-7.dll
- 2024-09-07  12:33 PM            24,064 libwebpdemux-2.dll
- 2024-09-07  12:33 PM           387,584 libxmp.dll
- 2024-09-07  12:33 PM    <DIR>          markupsafe
- 2024-09-07  12:33 PM    <DIR>          msgpack
- 2024-09-07  12:33 PM           827,728 MSVCR100.dll
- 2024-09-07  12:33 PM    <DIR>          multidict
- 2024-09-07  12:33 PM    <DIR>          numpy
- 2024-09-07  12:33 PM    <DIR>          numpy.libs
- 2024-09-07  12:33 PM    <DIR>          PIL
- 2024-09-07  12:33 PM            41,984 portmidi.dll
- 2024-09-07  12:33 PM    <DIR>          psutil
- 2024-09-07  12:33 PM    <DIR>          pycparser-2.22.dist-info
- 2024-09-07  12:33 PM           202,008 pyexpat.pyd
- 2024-09-07  12:33 PM    <DIR>          pygame
- 2024-09-07  12:33 PM    <DIR>          PyQt5
- 2024-09-07  12:33 PM            68,376 python3.dll
- 2024-09-07  12:33 PM         6,926,616 python312.dll
- 2024-09-07  12:33 PM    <DIR>          pythonnet
- 2024-09-07  12:33 PM    <DIR>          pywin32_system32
- 2024-09-07  12:33 PM         2,531,328 SDL2.dll
- 2024-09-07  12:33 PM           217,600 SDL2_image.dll
- 2024-09-07  12:33 PM           306,688 SDL2_mixer.dll
- 2024-09-07  12:33 PM         1,799,680 SDL2_ttf.dll
- 2024-09-07  12:33 PM            31,000 select.pyd
- 2024-09-07  12:33 PM    <DIR>          service_identity-24.1.0.dist-info
- 2024-09-07  12:33 PM    <DIR>          setuptools
- 2024-09-07  12:33 PM    <DIR>          setuptools-73.0.1.dist-info
- 2024-09-07  12:33 PM    <DIR>          tornado
- 2024-09-07  12:33 PM    <DIR>          typeguard-4.3.0.dist-info
- 2024-09-07  12:33 PM         1,016,584 ucrtbase.dll
- 2024-09-07  12:33 PM         1,138,456 unicodedata.pyd
- 2024-09-07  12:33 PM           119,192 VCRUNTIME140.dll
- 2024-09-07  12:33 PM            49,528 VCRUNTIME140_1.dll
- 2024-09-07  12:33 PM    <DIR>          webview
- 2024-09-07  12:33 PM    <DIR>          wheel-0.44.0.dist-info
- 2024-09-07  12:33 PM    <DIR>          win32
- 2024-09-07  12:33 PM    <DIR>          yaml
- 2024-09-07  12:33 PM    <DIR>          yarl
- 2024-09-07  12:33 PM           108,544 zlib1.dll
- 2024-09-07  12:33 PM    <DIR>          zope
- 2024-09-07  12:33 PM    <DIR>          zope.event-5.0.dist-info
- 2024-09-07  12:33 PM    <DIR>          zope.interface-6.4.post2.dist-info
- 2024-09-07  12:33 PM    <DIR>          zstandard
- 2024-09-07  12:33 PM            71,448 _asyncio.pyd
- 2024-09-07  12:33 PM           821,248 _brotli.cp312-win_amd64.pyd
- 2024-09-07  12:33 PM            85,272 _bz2.pyd
- 2024-09-07  12:33 PM           182,784 _cffi_backend.cp312-win_amd64.pyd
- 2024-09-07  12:33 PM           125,208 _ctypes.pyd
- 2024-09-07  12:33 PM           257,304 _decimal.pyd
- 2024-09-07  12:33 PM           133,400 _elementtree.pyd
- 2024-09-07  12:33 PM            66,328 _hashlib.pyd
- 2024-09-07  12:33 PM           160,024 _lzma.pyd
- 2024-09-07  12:33 PM            35,608 _multiprocessing.pyd
- 2024-09-07  12:33 PM            56,088 _overlapped.pyd
- 2024-09-07  12:33 PM            32,536 _queue.pyd
- 2024-09-07  12:33 PM            83,736 _socket.pyd
- 2024-09-07  12:33 PM           178,968 _ssl.pyd
- 2024-09-07  12:33 PM            25,880 _uuid.pyd
- 2024-09-07  12:33 PM            37,656 _wmi.pyd`

**Comment by brentvollebregt** [comment]
You could try adding some of these modules to `--exclude-module` (like `pygame`) to see if they get excluded. Also as recommended in the help post - using a virtual environment will help cut down this a lot as it will only have what you need in it + auto-py-to-exe.

**Comment by wgdesign44l** [comment]
Hello,
Please be kind to me. I'm a beginner and just need a single file called "S-Drive-Software V2.0.py" compiled into an .exe file for use on a Windows 11 Intel computer. I came across this page.
Can someone compile it for me if I provide the S-Drive software file?
That would be great, even if it seems a little bit rude.
Wolfgang

[auto-py-to-exe-master.zip](https://github.com/user-attachments/files/21974899/auto-py-to-exe-master.zip)

**Reply by brentvollebregt** [reply]
I highly recommend packaging the .py script to a .exe file yourself, as running .exes from people you don't know isn't a great idea.

Are you having issue using auto-py-to-exe to package your script?

**Reply by wgdesign44l** [reply]
Thank you for your reply. 
Yes, I'm having trouble because I don't have the appropriate environment to perform the conversion. So far, I've come across two additional programs that I would have to install and set up first. And all I need is this one .exe-file.
I need to modify an S-drive on a mobility device for seniors at a certain point in the access stream. A suitable programming device would unfortunately be too expensive at around €450.
